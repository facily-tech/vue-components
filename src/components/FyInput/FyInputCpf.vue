<template>
  <fy-input
    :value="text"
    @input="update"
    v-mask="mask"
    :rules="cpfRule"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </fy-input>
</template>

<script lang="ts">
import Vue from 'vue';
import { VueMaskDirective } from 'v-mask';
import FyInput from './FyInput.vue';

export default Vue.extend({
  name: 'fy-input-cpf',

  directives: { mask: VueMaskDirective },

  components: {
    FyInput,
  },

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      cpfRule: [
        (value: string): boolean | string =>
          value.length === 14 || 'O campo Cpf deve conter 11 n√∫meros',
      ],
      mask: '###.###.###-##',
    };
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
