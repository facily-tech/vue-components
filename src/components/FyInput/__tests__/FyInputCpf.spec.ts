import Vue from 'vue';
import Vuetify from 'vuetify';
import FyInputCpf from '../FyInputCpf.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyInputCpf.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInputCpf>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInputCpf, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
