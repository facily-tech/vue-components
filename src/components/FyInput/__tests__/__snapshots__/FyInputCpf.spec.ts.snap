// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInputCpf.ts should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined" value="" rules="(value) => {
          /* istanbul ignore next */
          cov_2m1k0a9qbu.f[2]++;
          cov_2m1k0a9qbu.s[9]++;
          return (
            /* istanbul ignore next */
            (cov_2m1k0a9qbu.b[1][0]++, value.length === 14) ||
            /* istanbul ignore next */
            (cov_2m1k0a9qbu.b[1][1]++, 'O campo Cpf deve conter 11 números')
          );
        }">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input data-test="input" id="input-3" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
